src/main/scala/backends/futil/FutilBackend.scala

```diff
@@ -774,14 +797,41 @@ private class FutilBackendHelper {
         case _ => store
       }
     )
-    val (cmdStruct, control, _) = emitCmd(p.cmd)(store)
+    val (cmdStruct, control, _) = emitCmd(p.cmd)(store, id2FuncDef)
+
+
+    val functionDefinitions: List[Component] =
+      for ((id, FuncDef(_, args, retType, Some(bodyOpt))) <- id2FuncDef.toList) yield {
+        val inputs = args.map(arg => PortDef(CompVar(arg.id.toString()), getBitwidth(arg.typ)))
+
+        val functionStore = inputs.foldLeft(Map[CompVar, CompVar]())((functionStore, inputs) =>
```

Comment:
> > There's one TODO that needs some more thinking.
> 
> Can you provide some context about this? I don't remember the details of `functionStore` etc.

Yep! Sorry, let me be more specific.

Let's say we have Dahlia function:
```
def foo(x: ubit<32>): ubit<32> {
  return x;
}
```
=>
```
import "primitives/std.lib";
component foo(x: 32) -> (out: 32) {
  cells {
    temp0 = prim std_reg(32);
  }
  wires {
    group let2<"static"=1> {
      temp0.in = x.out; // Uh-oh. We just want `x` here instead of `x.out`
      temp0.write_en = 1'd1;
      let2[done] = temp0.done;
    }
    out = temp0.out;
  }
  control {
    let2;
  }
}
...
```

So currently there exists [Store](https://github.com/cucapra/dahlia/blob/f76f6c1f415a6e9e5e1ea2cf396f534971f2d140/src/main/scala/backends/futil/FutilBackend.scala#L73) that maps Dahlia variables to Futil variables.

When walking over each component, I'm simply storing these into a Map `functionStore: Store` as `CompVar`s. However, what we actually want is to differentiate these since their port definitions for the given component. That means instead of `x.out`, we just need `x`.

An idea I mentioned earlier was to store these `PortDef`s in a different data structure, and pass them into `emit` functions. Here's a typical place where the changes would need to occur: [store.get()](https://github.com/cucapra/dahlia/blob/f76f6c1f415a6e9e5e1ea2cf396f534971f2d140/src/main/scala/backends/futil/FutilBackend.scala#L439)